src/factories/ClaimConditionFactory.ts

```diff
@@ -47,32 +49,37 @@ class ClaimConditionFactory {
    *
    * @returns - The claim conditions that will be used when validating a users claim transaction.
    */
-  public buildConditionsForDropV1(): PublicClaimCondition[] {
+  public async buildConditionsForDropV1(): Promise<PublicClaimCondition[]> {
     // v1 startTimestamp takes seconds from now.
     // v2 takes unix timestamp in seconds.
-    const publicClaimConditions = this.phases
-      .map((c) => c.buildPublicClaimCondition())
-      .map((c) => {
-        const now = Math.floor(Date.now() / 1000);
-        return {
-          ...c,
-          startTimestamp: c.startTimestamp.lt(now)
-            ? BigNumber.from(0)
-            : c.startTimestamp.sub(now),
-        };
+    let sorted: PublicClaimCondition[] = [];
+    await Promise.all(
+      this.phases.map(async (c) => await c.buildPublicClaimCondition()),
```

Comment:
you don't need `async (c) => await` here because `Promise.all` will resolve the promises in the array